File: mvvm-sample/src/main/java/com/boilerplate/mvvm_sample/ui/main/MainActivity.java
package com.boilerplate.mvvm_sample.ui.main;

import com.boilerplate.base.mvvm.BaseActivity;
import com.boilerplate.mvvm_sample.BR;
import com.boilerplate.mvvm_sample.R;
import com.boilerplate.mvvm_sample.databinding.ActivityMainBinding;

import javax.inject.Inject;

/**
 * Created by Louis on 2018/4/9.
 */

public class MainActivity extends BaseActivity<ActivityMainBinding, MainActivityViewModel> implements MainNavigator {

    @Inject
    MainActivityViewModel mainActivityViewModel;

    @Override
    public MainActivityViewModel getViewModel() {
        return mainActivityViewModel;
    }

    @Override
    public int getBindingVariable() {
        return BR.viewModel;
    }

    @Override
    public int getLayoutId() {
        return R.layout.activity_main;
    }
}

File: app-mvp-kotlin-sample/src/main/java/com/boilerplate/mvpSample/ui/main/MainActivity.kt
package com.boilerplate.mvpSample.ui.main

import com.boilerplate.base.kotlin.BaseActivity
import com.boilerplate.mvp_sample.R

/**
 * Created by Louis on 2018/5/1.
 */
class MainActivity : BaseActivity<MainPresenter, MainContract.View>(), MainContract.View {

    override var mView: MainContract.View
        get() = this
        set(value) {}
    override var mPresenter: MainPresenter
        get() = MainPresenter()
        set(value) {}
    override val layout: Int
        get() = R.layout.activity_main


    override fun showLoading() {
    }

    override fun completeLoading() {

    }

    override fun showMessage(msg: String) {

    }

    override fun getLayout() {

    }

    override fun setupActivityComponent() {

    }

    override fun initEventAndData() {

    }


}